catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -151,143 +162,134 @@ public void setAttributeOverrides(List<String> attributeOverrides) {
-                }
+    /*
+        Do not attempt to add routes to the CamelContext until we know the content scheme is ready.
+     */
+    private void waitForContentCamelComponent() throws InterruptedException {
+        final int maxRetries = 20;
+        final int sleepSeconds = 5;
+        int retryCount = 0;
+        while (camelContext.getComponent("content") == null) {
+            TimeUnit.SECONDS.sleep(sleepSeconds);
+            retryCount++;
+            if (retryCount > maxRetries) {
+                throw new IllegalStateException("Could not get Camel component 'content' ");
             }
-        } catch (Exception e) {
-            LOGGER.info("Unable to start Camel route with route ID = {}", routeId, e);
         }
     }
 
+    /*
+        Only remove routes that this Content Directory Monitor created since the same CamelContext
+        is shared across all Content Directory Monitors.
+     */
     private void removeRoutes() {
         LOGGER.trace("ENTERING: stopRoutes");
         List<RouteDefinition> routeDefinitions = camelContext.getRouteDefinitions();
         for (RouteDefinition routeDef : routeDefinitions) {
             try {
-                // Only remove routes that this Content Directory Monitor created
-                // (since same camelContext shared across all ContentDirectoryMonitors
-                // this is necessary)
-                if (isMyRoute(routeDef.getId())) {
-                    LOGGER.trace("Stopping route with ID = {}", routeDef.getId());
-                    camelContext.stopRoute(routeDef);  //DEPRECATED
-                    //                    camelContext.stopRoute(routeDef.getId());
-                    boolean status = camelContext.removeRoute(routeDef.getId());
-                    LOGGER.trace("Status of removing route {} is {}", routeDef.getId(), status);
+                String routeId = routeDef.getId();
+                if (isMyRoute(routeId)) {
+                    LOGGER.debug("Stopping route with ID = {}", routeId);
+                    camelContext.stopRoute(routeId);
+                    boolean status = camelContext.removeRoute(routeId);
+                    LOGGER.debug("Status of removing route {} is {}", routeId, status);
                     camelContext.removeRouteDefinition(routeDef);
                 }
             } catch (Exception e) {
-                LOGGER.info("Unable to stop Camel route with route ID = {}", routeDef.getId(), e);
+                LOGGER.warn("Unable to stop Camel route with route ID = {}", routeDef.getId(), e);
             }
         }
 
         LOGGER.trace("EXITING: stopRoutes");
     }
 
+    /*
+        Does the given routeId belong to this CDM's routeCollection?
+     */
     private boolean isMyRoute(String routeId) {
+        return this.routeCollection != null && this.routeCollection.stream()
+                .map(RouteDefinition::getId)
+                .anyMatch(s -> s.equals(routeId));
```

Comment:
I think you can do something like `.anyMatch(routeId::equals)` to use a method reference.